src/components/action-bar/action-bar.tsx

```diff
@@ -0,0 +1,283 @@
+     */
+    @Prop({ reflect: true })
+    public accessibleLabel?: string;
+
+    /**
+     * - When set to `fullWidth`, the component will take the
+     * entire width of its container.
+     * - When set to `floating`, the component will get basic stylings
+     * to visualize the floating state.
+     * :::note
+     * You should still properly position the component
+     * according to the structure of your user interface.
+     * For example, use an `absolute` or `fixed` position.
+     * :::
+     */
+    @Prop({ reflect: true })
+    public layout?: 'fullWidth' | 'floating';
+
+    /**
+     * Fired when a action bar item has been clicked.
+     */
+    @Event()
+    public itemSelected: EventEmitter<ActionBarItem>;
+
+    @Element()
+    public element: HTMLElement;
+
+    /**
+     * Defines the location that the content of the overflow menu
+     * appears, in relation to its trigger.
+     */
+    @Prop({ reflect: true })
+    public openDirection: OpenDirection;
+
+    @State()
+    private overflowCutoff: number = this.actionBarItems?.length || 0;
+
+    @State()
+    private calculatingOverflow: boolean = false;
+
+    private resizeObserver: ResizeObserver;
+    private intersectionObserver: IntersectionObserver;
+    private overflowRecalculationScheduled: boolean;
+    // eslint-disable-next-line no-magic-numbers
+    private RESIZE_RATE_MS = 50;
+    // eslint-disable-next-line no-magic-numbers
+    private RESIZE_TIMEOUT = 250;
+    private lastOverflowCutoff: number = this.actionBarItems?.length || 0;
+    private lastResize = Date.now();
+
+    public componentWillLoad() {
+        this.resizeObserver = new ResizeObserver(this.handleResize);
+    }
+
+    public disconnectedCallback() {
+        this.resizeObserver.disconnect();
+        this.intersectionObserver.disconnect();
+    }
+
+    public render() {
+        let visibleItems: Array<ActionBarItem | ListSeparator> = [];
+        let overflowItems: Array<MenuItem | ListSeparator> = [];
+        if (this.actionBarItems.length) {
+            visibleItems = this.actionBarItems.slice(0, this.overflowCutoff);
+            overflowItems = this.actionBarItems.slice(this.overflowCutoff);
+        }
+
+        return (
+            <Host
+                aria-label={this.accessibleLabel}
+                class={{
+                    'is-full-width': this.layout === 'fullWidth',
+                    'is-floating': this.layout === 'floating',
+                }}
+            >
+                {visibleItems.map(this.renderActionBarItem)}
+                {this.renderOverflowMenu(overflowItems)}
+            </Host>
+        );
+    }
+
+    public componentDidRender() {
+        const element =
+            this.element.parentElement ?? (this.element.parentNode as any).host;
+
+        if (element === null) {
+            throw new Error(
+                'Failed to create resizeObserver, no parent found for limel-action-bar. Consider wrapping the action-bar in an element'
```

Comment:
This seems to me like a curious error message… When will this happen? When someone uses document.createElement?

Reply:
I think it's due to this issue :smile: https://github.com/Lundalogik/lime-elements/pull/2315#discussion_r1214217958

The consumer is forced to put the element inside a `div` or similar (something without a shadow DOM). But we should fix that since it's a weird restriction :slightly_smiling_face:

Reply:
What @jgroth said 🙂 

At first we didn't know how to get some element to pass on to the resize observer if the parent element was a shadow dom. But it's solved now.